processor/groupbytraceprocessor/factory.go

```diff
@@ -0,0 +1,100 @@
+// Copyright The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package groupbytraceprocessor
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	"go.opentelemetry.io/collector/component"
+	"go.opentelemetry.io/collector/config/configerror"
+	"go.opentelemetry.io/collector/config/configmodels"
+	"go.opentelemetry.io/collector/consumer"
+)
+
+const (
+	// typeStr is the value of "type" for this processor in the configuration.
+	typeStr configmodels.Type = "groupbytrace"
+)
+
+var (
+	defaultWaitDuration   = time.Second
+	defaultNumTraces      = 1_000_000
+	defaultDiscardOrphans = false
+	defaultStoreOnDisk    = false
+
+	errDiskStorageNotSupported    = fmt.Errorf("option 'disk storage' not supported in this release")
+	errDiscardOrphansNotSupported = fmt.Errorf("option 'discard orphans' not supported in this release")
+)
+
+// Factory is the factory for the processor.
+type Factory struct {
+}
+
+var _ component.ProcessorFactory = (*Factory)(nil)
```

Comment:
Please use the helper factory :), reduces the code and makes things consistent

Reply:
Done!